### PR TITLE
gamm: Use expectedErr fields with error types

### PR DESCRIPTION
**Pull request by hieuvubk**
Partially closes: #2959

## What is the purpose of the change

> Update tests that are using expectPass or expectErr as the bool variable to the error type in the gamm module

## Brief Changelog

*(for example:)*
 
  - *Coverage gamm module*
  - *All tests pass*
  - *Commented at test cases where the error returned is not as expected*

### CODE REVIEW

File: x/gamm/keeper/msg_server_test.go [hunk 1]
```diff
@@ -305,7 +316,7 @@ func (suite *KeeperTestSuite) TestExitPool_Events() {
 			poolId:        1,
 			shareInAmount: sdk.NewInt(shareIn),
 			tokenOutMins:  sdk.NewCoins(sdk.NewCoin("foo", sdk.NewInt(tokenOutMinAmount))),
-			expectError:   true,
+			expectError:   types.ErrLimitMinAmount,
```
Comment (mattverse) [inline x/gamm/keeper/msg_server_test.go:319]: Isn't this error returned wrapped with the sdk error? e.g `sdkerrors.Wrapf(types.ErrLimitMaxAmount, sharesLargerThanMaxErrFormat, sharesIn.Int64(), shareInMaxAmount.Uint64())` ?

File: x/gamm/keeper/pool_service_test.go [hunk 2]
```diff
@@ -419,7 +418,7 @@ func (suite *KeeperTestSuite) TestJoinPoolNoSwap() {
 			tokenInMaxs: sdk.Coins{
```
Comment (mattverse) [inline x/gamm/keeper/pool_service_test.go:418]: Ah this test case was wrong, glad that we have this PR + change. This shouldn't be testing with negative shares, it should have positivie shares and we're expecting error related to insufficient funds, not negative share ratio. Would you be able to change this test case within this PR as well? We can also do it in a separate PR if you would prefer that way!

**Review by mattverse — COMMENTED** [review]
Looks nice! Left some few comments that I think should be addressed before merging!

File: x/gamm/keeper/pool_service_test.go (hunk 2 above)
Comment (hieuvubk) [reply x/gamm/keeper/pool_service_test.go]: I updated in this pr

File: x/gamm/keeper/msg_server_test.go (hunk 1 above)
Comment (hieuvubk) [reply x/gamm/keeper/msg_server_test.go]: ah missed, updated

**Comment by czarcas7ic** [comment]
Copying my previous message here:

There seems to be some miscommunication internally on desired endstate for tests. While you completed what was requested from the github issue, we had internally decided to stick with bool error checking (for now) and possibly discuss moving to this at a later time. cc @mattverse

**Comment by czarcas7ic** [comment]
Going to close this for now and we can potentially reopen after further discussions
